### ISSUE TITLE
Configuration directive for how many clamav scans to run in paralel.

### ISSUE THREAD

**Issue by a-rehak** [owncloud/files_antivirus#485]
Hello.

I am not sure if there was a proposal of this kind in the past - i have tried to search for, but found nothing - in any case: I would like to propose an enhancement feature.
It would be really nice to have a configuration directive that would limit the amount of spawned `clamscan` processes that are running in parallel.

The reasoning for this is that we all know that clamav is very resource (RAM) hungry and CPU intensive process.
So if there is anybody, like me, running owncloud on a low resource machine and with clamav integration.
Somebody uploads 40 files and each one of them are started to be scanned in parallel by clamav.
It would be nice to form a _queue_ and using a config directive like (for example) `max_parallel_clamav_processes`, spawn only that many amount of processes - effectively putting a limit to memory and number of CPU cores utilized.

**Another suggestion**:
A nice to have feature would also be the ability to prefix the _CLI call_ to clamscan  with a custom command. For example: `cpulimit  -l 30` or `nice -n 19`.

Thank you for this marvelous feature.
Please let me know what you think.

**Comment by DeepDiver1975** [comment]
scanning via clamav cli is quite time consuming an resource hungry for sure.

I suggest to use ClamAV via tcp or unix socket.
